addons/account/models/res_partner_bank.py

```diff
@@ -285,7 +285,7 @@ def write(self, vals):
         if ('acc_number' in vals or 'partner_id' in vals) and any(account.lock_trust_fields for account in self):
             raise UserError(_("You cannot modify the account number or partner of an account that has been trusted."))
 
-        if 'allow_out_payment' in vals and not self.user_has_groups('account.group_validate_bank_account'):
+        if 'allow_out_payment' in vals and not (self.user_has_groups('account.group_validate_bank_account') or self.env.is_superuser()):
```

Comment:
@qdp-odoo I don't understand the implementation.
Why don't we allow anybody to untrust?

```suggestion
        vals.get('allow_out_payment') and not self.user_has_groups('account.group_validate_bank_account'):`
```

Reply:
As stated in the commit message, a change in the address_home_id with a recent fix would lead to the change of `allow_out_payment`. If an employee goes through the salary configurator, their data will always be updated. As all employee don't have those rights, it leads to an issue:
https://github.com/odoo/odoo/blob/master/addons/hr/models/hr_employee.py#L395
The change was made in this commit: https://github.com/odoo/odoo/commit/e89c54b1fe48804083b9704d4370ce89b261b340 though it seems that it was already trying to avoid access issues.
Instead, I'll just add a check on the employee write method to avoid writing that field if it's unnecessary
```diff
-                    bank_account.sudo().allow_out_payment = False
+                   if bank_account.allow_out_payment:
+                       bank_account.sudo().allow_out_payment = False
```
This fixes my test though I don't now if it will cover every possible case (though the PR was merged so I guess that every other test work fine)